logprep/processor/amides/processor.py

```diff
@@ -0,0 +1,201 @@
+        cache_load: float = 0.0
+        """Relative cache load."""
+        mean_misuse_detection_time: float = 0.0
+        """Mean processing time of command-lines classified by the misuse detection model."""
+        _mean_misuse_detection_time_sample_counter: int = 0
+        mean_rule_attribution_time: float = 0.0
+        """Mean processing time of command-lines classified by the rule attribution model."""
+        _mean_rule_attribution_time_sample_counter: int = 0
+
+        def update_mean_misuse_detection_time(self, new_sample):
+            """Updates the mean processing time of the misuse detection."""
+            new_mean, new_sample_counter = calculate_new_average(
+                self.mean_misuse_detection_time,
+                new_sample,
+                self._mean_misuse_detection_time_sample_counter,
+            )
+            self.mean_misuse_detection_time = new_mean
+            self._mean_misuse_detection_time_sample_counter = new_sample_counter
+
+        def update_mean_rule_attribution_time(self, new_sample):
+            """Updates the mean processing time of the rule attribution model."""
+            new_mean, new_sample_counter = calculate_new_average(
+                self.mean_rule_attribution_time,
+                new_sample,
+                self._mean_rule_attribution_time_sample_counter,
+            )
+            self.mean_rule_attribution_time = new_mean
+            self._mean_rule_attribution_time_sample_counter = new_sample_counter
+
+    __slots__ = ["_cache", "_misuse_detector", "_rule_attributor", "_normalizer"]
+
+    _cache: LRUCache
+    _misuse_detector: MisuseDetector
+    _rule_attributor: RuleAttributor
+    _normalizer: CommandLineNormalizer
+
+    rule_class = AmidesRule
+
+    def __init__(self, name: str, configuration: Processor.Config, logger: Logger):
+        super().__init__(name=name, configuration=configuration, logger=logger)
+        self.metrics = self.AmidesMetrics(
+            labels=self.metric_labels,
+            generic_rule_tree=self._generic_tree.metrics,
+            specific_rule_tree=self._specific_tree.metrics,
+        )
+        self._cache = LRUCache(max_items=self._config.max_cache_entries)
+        self._normalizer = CommandLineNormalizer(max_num_values_length=3, max_str_length=30)
+        self._setup_detectors()
+
+    def _setup_detectors(self):
+        models = self._load_models(self._config.models_path)
+
+        self._misuse_detector = MisuseDetector(models["single"], self._config.decision_threshold)
+        self._rule_attributor = RuleAttributor(
+            models["multi"],
+            self._config.num_rule_attributions,
+        )
+
+    def _load_models(self, path):
+        models_filename = basename(path).rstrip(".zip")
+
+        with ZipFile(path, mode="r") as zip_file:
+            with zip_file.open(models_filename, mode="r") as models_file:
+                models = joblib.load(models_file)
```

Comment:
I have made changes so that models can be loaded via HTTP, or a local copy is being created, via the `Getter` interface, depending on the URI/path in the processor's configuration.

Models provided to the processor can be quite large. Thus, using a compressed format appears as the most practical solution to me.

The models we currently use are generated via `scikit-learn`, and need to be pickled in order to be used by the processor.  Pickling these models can be done via `joblib` or `pickle`. As `joblib` is required by `scikit-learn` anyways,  we decided to stay with `joblib` for now. If there are any other concerns, let me know.

Reply:
what I see is great! thx